src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Simple.cs

```diff
@@ -0,0 +1,573 @@
+
+                server.Disconnect();
+
+                Assert.Throws<InvalidOperationException>(() => server.Disconnect());    // double disconnect
+
+                byte[] buffer = new byte[] { 0, 0, 0, 0 };
+
+                if (pair.writeToServer)
+                {
+                    Assert.Throws<InvalidOperationException>(() => server.Write(buffer, 0, buffer.Length));
+                    Assert.Throws<InvalidOperationException>(() => server.WriteByte(5));
+                    Assert.Throws<InvalidOperationException>(() => { server.WriteAsync(buffer, 0, buffer.Length); });
+                }
+                else
+                {
+                    Assert.Throws<InvalidOperationException>(() => server.Read(buffer, 0, buffer.Length));
+                    Assert.Throws<InvalidOperationException>(() => server.ReadByte());
+                    Assert.Throws<InvalidOperationException>(() => { server.ReadAsync(buffer, 0, buffer.Length); });
+                }
+
+                Assert.Throws<InvalidOperationException>(() => server.Flush());
+                Assert.Throws<InvalidOperationException>(() => server.IsMessageComplete);
+            }
+        }
+
+        [Fact]
+        public virtual async Task OperationsOnDisconnectedClient()
+        {
+            using (var pair = CreateNamedPipePair())
+            {
+                var server = pair.serverStream;
+                var client = pair.clientStream;
+                pair.Connect();
+
+                Assert.Throws<InvalidOperationException>(() => client.IsMessageComplete);
+                Assert.Throws<InvalidOperationException>(() => client.Connect());
+                await Assert.ThrowsAsync<InvalidOperationException>(() => client.ConnectAsync());
+
+                server.Disconnect();
+
+                byte[] buffer = new byte[] { 0, 0, 0, 0 };
+
+                if (!pair.writeToServer)
+                {
+                    // Pipe is broken
+                    Assert.Throws<IOException>(() => client.Write(buffer, 0, buffer.Length));
+                    Assert.Throws<IOException>(() => client.WriteByte(5));
+                    Assert.Throws<IOException>(() => { client.WriteAsync(buffer, 0, buffer.Length); });
+                    Assert.Throws<IOException>(() => client.Flush());
+                }
+                else
+                {
+                    // Nothing for the client to read, but no exception throwing
+                    Assert.Equal(0, client.Read(buffer, 0, buffer.Length));
+                    Assert.Equal(-1, client.ReadByte());
+                }
+
+                Assert.Throws<InvalidOperationException>(() => client.IsMessageComplete);
+            }
+        }
+
+        [Fact]
+        public void OperationsOnUnconnectedServer()
+        {
+            using (var pair = CreateNamedPipePair())
+            {
+                var server = pair.serverStream;
```

Comment:
There's a fair amount of ```var``` being used when the type isn't clear from the right-hand side... can we change those to use the actual type name?